source/Launchpad-Applications/CurrentlyRunningLaunchpadApplication.class.st

```diff
@@ -0,0 +1,60 @@
+Class {
+	#name : 'CurrentlyRunningLaunchpadApplication',
+	#superclass : 'Object',
+	#instVars : [
+		'binding'
+	],
+	#classInstVars : [
+		'uniqueInstance'
+	],
+	#category : 'Launchpad-Applications',
+	#package : 'Launchpad-Applications'
+}
+
+{ #category : 'initialization' }
+CurrentlyRunningLaunchpadApplication class >> initialize [
+
+	<ignoreForCoverage>
+	self initializeUniqueInstance
+]
+
+{ #category : 'initialization' }
+CurrentlyRunningLaunchpadApplication class >> initializeUniqueInstance [
+
+	uniqueInstance := super new
+]
+
+{ #category : 'instance creation' }
+CurrentlyRunningLaunchpadApplication class >> new [
+
+	^uniqueInstance
+]
+
+{ #category : 'accessing' }
+CurrentlyRunningLaunchpadApplication >> currentlyRunning [
+
+	binding ifNil: [ self resetCurrentlyRunning ].
+	^ binding content
+]
+
+{ #category : 'accessing' }
+CurrentlyRunningLaunchpadApplication >> resetCurrentlyRunning [
+
+	binding := Binding undefinedExplainedBy:
+		               'There''s no current application running.'
+]
+
+{ #category : 'accessing' }
+CurrentlyRunningLaunchpadApplication >> setAsCurrentlyRunning: anApplication [
+
+	binding := Binding to: anApplication
+]
+
+{ #category : 'accessing' }
+CurrentlyRunningLaunchpadApplication >> setAsCurrentlyRunning: anApplication during: aBlock [
```

Comment:
```suggestion
CurrentlyRunningLaunchpadApplication >> set :anApplication  asCurrentlyRunningDuring: aBlock [
```

Reply:
Agreed on not to rename this method for consistency with the other variant without during: aBlock